variables.tf

```diff
@@ -210,3 +210,13 @@ variable "log_retention_days" {
   type        = string
   default     = "0"
 }
+
+variable "runtime_platform" {
+  description = "The runtime platform configuration"
+  type = object({
+      cpu_architecture = string
+    })
+    default = ({
+      cpu_architecture = "X86_64"
+    })
+}
```

Comment:
I suggest to add a validation to allow only the two values that we know we can use here:
```
  validation {
    condition     = contains(["X86_64", "ARM64"], var.runtime_platform)
    error_message = "Allowed values for runtime_platform are \"X86_64\" or \"ARM64\"."
  }
```

Reply:
Oh, nice, thx

Reply:
Fixed by [24e0b77](https://github.com/sysdiglabs/terraform-aws-fargate-orchestrator-agent/pull/30/commits/24e0b772e6be100ea248b1a5f1ded49e1fac3509)

### on `ARM`

```hcl
$ terraform apply
var.cpu_architecture
  Enter a value: ARM64

...

  # module.sysdig_orchestrator_agent.aws_ecs_task_definition.orchestrator_agent will be created
  + resource "aws_ecs_task_definition" "orchestrator_agent" {
      ...

      + runtime_platform {
          + cpu_architecture = "ARM64"
        }
    }
```

### on `X86_64`

```hcl
$ terraform apply
var.cpu_architecture
  Enter a value: X86_64

...

  # module.sysdig_orchestrator_agent.aws_ecs_task_definition.orchestrator_agent will be created
  + resource "aws_ecs_task_definition" "orchestrator_agent" {
      ...

      + runtime_platform {
          + cpu_architecture = "X86_64"
        }
    }
```

### on `potato`

```hcl
$ terraform apply
var.cpu_architecture
  Enter a value: potato

...

Plan: 16 to add, 0 to change, 0 to destroy.
╷
│ Error: Invalid value for variable
│
│   on main.tf line 12, in module "sysdig_orchestrator_agent":
│   12:   runtime_platform = {
│   13:     cpu_architecture = var.cpu_architecture
│   14: }
│     ├────────────────
│     │ var.runtime_platform.cpu_architecture is "potato"
│
│ The runtime_platform.cpu_architecture must be either 'ARM64' or 'X86_64'
│
│ This was checked by the validation rule at ../variables.tf:222,3-13.
```